### PR TITLE
PUT Endpoint

### PR DESCRIPTION
**Pull request by jeethsoni**
Added `PUT` blueprint and service to update a record by `movie_id`

### CODE REVIEW

File: blueprints/movie/blueprint.py [hunk 1]
```diff
@@ -47,3 +47,21 @@ def post_movie():
...
+    """
+    A PUT handler.
+    Updates a record by id
+    """
+
+    # request object
+    payload = request.get_json()
+
+    # primary key of the table
+    pk = movie_id
```
Comment (riteshapatel) [inline blueprints/movie/blueprint.py]: Why this?

File: blueprints/movie/blueprint.py [hunk 2]
```diff
@@ -47,3 +47,21 @@ def post_movie():
...
+
+    result = svc_put(payload, pk)
```
Comment (riteshapatel) [inline blueprints/movie/blueprint.py]: Pk can be straight up movie_id no?

File: blueprints/movie/service.py [hunk 3]
```diff
@@ -46,3 +45,26 @@ def svc_post(payload):
...
+    year = payload["movie_year"]
+    rating = payload["rating"]
+    runtime = payload["runtime"]
+    votes = payload["votes"]
+    revenue = payload["revenue"]
+    metascore = payload["metascore"]
+    created_at = payload["created_at"]
+
+    # SQL statement
+    sql = f"UPDATE {SCHEMA_NAME}.{MOVIE} SET title = %(title)s, description = %(description)s, movie_year = %(movie_year)s, rating = %(rating)s, runtime = %(runtime)s, votes = %(votes)s, revenue = %(revenue)s, metascore = %(metascore)s, created_at = %(created_at)s WHERE movie_id = %(id)s RETURNING *;"
```
Comment (riteshapatel) [inline blueprints/movie/service.py]: Format SQL properly so it’s readable

File: blueprints/movie/service.py [hunk 4]
```diff
@@ -46,3 +45,26 @@ def svc_post(payload):
...
+    params = {"title": title, "description": description, "movie_year": year, "rating": rating, "runtime": runtime, "votes": votes, "revenue": revenue, "metascore": metascore, "created_at": created_at, "id": id}
```
Comment (riteshapatel) [inline blueprints/movie/service.py]: Why dictionary is not prettified? Make it readable. VScode can format it for ya! Or use jsonlint to format it. Google it.

**Review by riteshapatel — CHANGES_REQUESTED** [review]

File: blueprints/movie/blueprint.py (hunk 1 above)
Comment (jeethsoni) [reply blueprints/movie/blueprint.py]: pk = movie?

File: blueprints/movie/service.py [hunk 5]
```diff
@@ -89,3 +89,31 @@ def svc_put(payload, id):
...
+    """
+    A DELETE service
+    """
+
+    # deletes from the child table first and then deletes from parent table
+    sql = (
+        # child table
+        """
+        DELETE FROM movies.movie_review WHERE movie_id = %(movie_id)s;
+        """
```
Comment (riteshapatel) [inline blueprints/movie/service.py]: ☝️ Remove these comments. SQL as a comment?  Also where are you executing this SQL? Code is wrong.

File: blueprints/movie/service.py [hunk 6]
```diff
@@ -89,3 +89,31 @@ def svc_put(payload, id):
...
+        RETURNING *;
+        """
+        )
+
+    params = {
+            "movie_id": movie_id
+            }
+
+    result = do_query(sql, params)
+    print(result)
```
Comment (riteshapatel) [inline blueprints/movie/service.py]: Remove `print` statements before you push. These `print` statements would make it into `production`. Not good.

**Review by riteshapatel — CHANGES_REQUESTED** [review]
Sending back for changes.

**Review by riteshapatel — APPROVED** [review]

**Pull request merged**